### PR TITLE
SALTO-1660 updated config

### PR DESCRIPTION
**Pull request by dantal4**
updated config

---

_Additional context for reviewer_
this is the first PR for this ticket, followed by [https://github.com/salto-io/salto/pull/2500](url)

---
_Release Notes_: 
Zuora Adapter:
- added `settingsSwagger: { typeNameOverrides: [...] }` to ZuoraApiConfig to be able to override typeNames in "settings" types

---
_User Notifications_: 
- hide `id` field in fetched instances.

### CODE REVIEW

File: packages/zuora-billing-adapter/src/transformers/billing_settings.ts [hunk 1]
```diff
@@ -88,9 +88,16 @@ export const generateBillingSettingsTypes = async (
   )
 
   const schemasAndRefs = toSchemasAndRefs(settingsInfos)
+  const settingsApiDefConfig = {
+    ...apiDefConfig,
+    swagger: {
+      url: apiDefConfig.swagger.url,
```
Comment (netama) [inline packages/zuora-billing-adapter/src/transformers/billing_settings.ts:94]: nit: please explain in a comment that this arg will be ignored (right?)

File: packages/zuora-billing-adapter/test/adapter.test.ts [hunk 2]
```diff
@@ -285,10 +285,12 @@ describe('adapter', () => {
           'Settings_SubscriptionSetting',
           'Settings_TaxCode',
           'Settings_TaxCompany',
+          'Settings_TaxEngine',
+          'Settings_TaxEngines',
```
Comment (netama) [inline packages/zuora-billing-adapter/test/adapter.test.ts:289]: do we need both?

File: packages/zuora-billing-adapter/src/config.ts [hunk 3]
```diff
@@ -275,16 +292,49 @@ const DEFAULT_TYPE_CUSTOMIZATIONS: ZuoraApiConfig['types'] = {
...
+  [`${SETTINGS_TYPE_PREFIX}SegmentationRule`]: {
+    transformation: {
+      fieldsToHide: [
+        { fieldName: 'id', fieldType: 'string' },
+      ],
+    },
+  },
+  [`${SETTINGS_TYPE_PREFIX}RuleDetail`]: {
+    transformation: {
+      fieldTypeOverrides: [
```
Comment (netama) [inline packages/zuora-billing-adapter/src/config.ts:330]: continuing from https://github.com/salto-io/salto/pull/2500/files#r759469785 - I don't see that we have an example of this in our fetch, and I want to make sure we don't create explicit fields for something that may not always be available - let's discuss this one IRL :)

File: packages/zuora-billing-adapter/src/config.ts [hunk 4]
```diff
@@ -275,16 +292,49 @@ const DEFAULT_TYPE_CUSTOMIZATIONS: ZuoraApiConfig['types'] = {
...
+        { fieldName: 'segmentId', fieldType: 'string' },
+        { fieldName: 'segmentName', fieldType: 'string' },
+        { fieldName: 'transactionType', fieldType: 'string' },
+      ],
+      fieldsToOmit: [
+        { fieldName: 'segmentId' },
```
Comment (netama) [inline packages/zuora-billing-adapter/src/config.ts:336]: why omit and not hide? can this be nested?

**Review by netama — COMMENTED** [review]
looks good, with a couple of questions (let's discuss IRL and then this should be ok to merge)

File: packages/zuora-billing-adapter/test/adapter.test.ts (hunk 2 above)
Comment (dantal4) [reply packages/zuora-billing-adapter/test/adapter.test.ts:289]: - in `config.ts` we fetch `Settings_TaxEngines` only
- its content is `{ taxEngines: list<Settings_TaxEngine> }`
- `taxEngines` defined as standalone fields

- therefore, they are in elements too

**Review by netama — APPROVED** [review]

**Pull request merged**